### DISCUSSION TITLE
How can I open files from local place in C++/WinRT?

### DISCUSSION THREAD

**Discussion by Extremeer** [microsoft/WindowsAppSDK#3164] [Q&A] [answered]
It seems that OPENFILENAME and GetOpenFileName cannot work correctly.
Error: unresolved external symbol __imp_GetOpenFileNameW,function "int __cdecl winrt::ProjectName
OPENFILENAME ofn;
ZeroMemory(&ofn, sizeof(ofn));
ofn.lStructSize = sizeof(ofn);
ofn.lpstrFile = path;
ofn.nMaxFile = MAX_PATH;
ofn.lpstrFilter = "txt\0*.TXT\0"; // ofn.lpstrFilter = "All\0*.*\0Text\0*.TXT\0";
ofn.Flags = OFN_PATHMUSTEXIST | OFN_FILEMUSTEXIST;
return GetOpenFileName(&ofn);

**Comment by DarranRowe** [comment, marked as answer]
For GetOpenFileName, you need to link with comdlg32.lib, if you don't then you will get an unresolved external symbol error.

**Reply by Extremeer** [reply]
Thanks!
But after that, there’s another error: A value of type 'char *' cannot be assigned to a real of type 'LPWSTR'.
I cannot find any property in the project property to change character set into ANSI just like MFC.

**Reply by DarranRowe** [reply]
The versions without a suffix were always intended to be used as a generic version. To properly use them, you would have to use the TCHAR character type to define the string and the TEXT macro to properly prefix the string literals.
If you really want to/must use the ANSI version, then your best option is to explicitly use the OPENFILENAMEA structure and call the GetOpenFileNameA function.

```
std::string path = "Test.txt";

OPENFILENAMEA ofn{sizeof(OPENFILENAMEA)};
ofn.lpstrFile = path.data();
ofn.nMaxFile = MAX_PATH;
ofn.lpstrFilter = "All\0*.*\0Text\0*.txt";
ofn.Flags = OFN_PATHMUSTEXIST | OFN_FILEMUSTEXIST;

GetOpenFileNameA(&ofn);
```

**Reply by Extremeer** [reply]
Thanks! Got it!
